lib/rex/post/meterpreter/extensions/stdapi/fs/file.rb

```diff
@@ -142,8 +142,7 @@ def File.expand_path(path)
         # First check for ~
         path_components = path.split(separator)
         if path_components.length > 0 && path_components[0] == '~'
-          path_components[0] = '$HOME'
-          path = path_components.join(separator)
+          path = "$HOME#{path[1..-1]}"
```

Comment:
Actually this is entirely my fault, as even if it is `nil` it will get translated into an empty string by the string interpolation. Ignore what I said above, marking this as resolved.